internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -178,13 +179,20 @@ func newEnrollCmd(
 	options *enrollCmdOption,
 	configPath string,
 	store saver,
+	backoffFactory func(done <-chan struct{}) backoff.Backoff,
```

Comment:
nit; I understand that this might be a personal preference but the "Factory" here doesn't seem 100% appropriate to me. I may be missing something but AFAIK the field `backoffFactory` can seamlessly become `backoff backoff.Backoff` and then inside `enrollWithBackoff` if it is nil assign it to `backoff.NewEqualJitterBackoff(signal, enrollBackoffInit, enrollBackoffMax)`. In this way we keep "close" at the code the signal that this backoff depends on and we don't hold a ref from a function that the enrollCmd instantiator might provide. No strong feelings, just as an extra approach here 🙂

Reply:
Sorry my intention wasn't to delay this PR from getting merged. I was initially under the impression that we use as signal the `ctx context.Context` passed in `enrollWithBackoff` thus I believed that since the caller is giving us this context it can instantiate properly the backoff and pass it as a parameter in this function as well; I don't see any point having the backoff inside enrollCmd struct as it is used only in `enrollWithBackoff`. However, after looking again at the code, I did notice that we use a `chan struct{}` as the signal thus Factory makes more sense now to me. Resolving this no need to do anything about my comment in this PR 🙂